File: app/src/main/java/com/philkes/pin2pdf/model/PinModel.java
package com.philkes.pin2pdf.model;

import org.apache.commons.lang3.StringEscapeUtils;

public class PinModel {
    private Integer id;
    private String pinId;
    private String title;
    private String imgUrl;
    private String link;
    private String pdfLink;
    private String board;
    private String note;

    public PinModel(Integer id, String pinId, String title, String imgUrl, String link,
                    String pdfLink, String board,
                    String note) {
        this.id=id;
        this.pinId=pinId;
        setTitle(title);
        setImgUrl(imgUrl);
        setLink(link);
        setPdfLink(pdfLink);
        setBoard(board);
        setNote(note);
    }

    public String getBoard() {
        return board;
    }

    public void setBoard(String board) {
        this.board=StringEscapeUtils.unescapeHtml3(board)
                .replace("/", "")
                .replace(" ", "-");
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title=StringEscapeUtils.unescapeHtml3(title);
    }

    public String getImgUrl() {
        return imgUrl;
    }

    public void setImgUrl(String imgUrl) {
        this.imgUrl=imgUrl;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link=link;
    }

    public String getPdfLink() {
        return pdfLink;
    }

    public void setPdfLink(String pdfLink) {
        this.pdfLink=pdfLink;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id=id;
    }

    public String getNote() {
        return note;
    }

    public void setNote(String note) {
        this.note=note;
    }

    public String getPinId() {
        return pinId;
    }

    public void setPinId(String pinId) {
        this.pinId=pinId;
    }
}

File: app/src/main/java/com/philkes/pin2pdf/model/PinModel.kt
package com.philkes.pin2pdf.model

import org.apache.commons.lang3.StringEscapeUtils

/**
 * DTO Object for [com.philkes.pin2pdf.storage.local.entity.Pin] Entity
 */
class PinModel(
    var id: Int?, var pinId: String?, mTitle: String?, mImgUrl: String?, mLink: String?,
    mPdfLink: String?, mBoard: String?,
    mNote: String?
) {
    var title: String? = null
        set(title) {
            field = StringEscapeUtils.unescapeHtml3(title)
        }
    var imgUrl: String? = null
    var link: String? = null
    var pdfLink: String? = null
    var board: String? = null
        set(board) {
            field = StringEscapeUtils.unescapeHtml3(board)
                .replace("/", "")
                .replace(" ", "-")
        }
    var note: String? = null

    init {
        title = mTitle
        imgUrl = mImgUrl
        link = mLink
        pdfLink = mPdfLink
        board = mBoard
        note = mNote
    }
}